src/graph/entities/edge.h

```diff
@@ -15,15 +15,17 @@
 /* TODO: note it is possible to get into an inconsistency
  * if we set src and srcNodeID to different nodes. */
 struct Edge {
-    Entity *entity;          /* MUST be the first property of Edge. */
-    char *alias;             /* Alias attached to edge. */
-    char* relationship;      /* Label attached to edge. */
-    int relationID;          /* Relation ID. */
-    Node* src;               /* Pointer to source node. */
-    Node* dest;              /* Pointer to destination node. */
-    NodeID srcNodeID;        /* Source node ID. */
-    NodeID destNodeID;       /* Destination node ID. */
-    GrB_Matrix mat;          /* Adjacency matrix, associated with edge. */
+    Entity *entity;         /* MUST be the first property of Edge. */
+    char *alias;            /* Alias attached to edge. */
+    char* relationship;     /* Label attached to edge. */
+    int relationID;         /* Relation ID. */
+    Node* src;              /* Pointer to source node. */
+    Node* dest;             /* Pointer to destination node. */
+    NodeID srcNodeID;       /* Source node ID. */
+    NodeID destNodeID;      /* Destination node ID. */
+    GrB_Matrix mat;         /* Adjacency matrix, associated with edge. */
+    unsigned int minHops;   /* Minimum number of hops this edge represents. */
+	unsigned int maxHops;   /* Maximum number of hops this edge represents. */
```

Comment:
I think we should consider having different Node and Edge structs for the QueryGraph than we do for actual data, since these variables don't make sense in the data context, but increase the size of each Record Entry by 8 bytes. Additionally, this duality muddles things like multiple relation types, which are valid in the QueryGraph but not in the data. This is handled in querying properly by adding matrices, but we can see it mess up in the EXPLAIN logic:
```
$ redis-cli GRAPH.EXPLAIN social "MATCH (a)-[e:cc|:dd]->(b) RETURN a, e, b"
1) "Results"
2) "    Project"
3) "        Conditional Traverse | (a)-[e:cc]->(b)"
4) "            All Node Scan | (a)"
```